packages/vite/src/node/config.ts

```diff
@@ -768,11 +770,21 @@ export function mergeConfig(
   return mergeConfigRecursively(defaults, overrides, isRoot ? '' : '.')
 }
 
-function mergeAlias(a: AliasOptions = [], b: AliasOptions = []): Alias[] {
-  return [...normalizeAlias(a), ...normalizeAlias(b)]
+function mergeAlias(
+  a?: AliasOptions,
+  b?: AliasOptions
+): AliasOptions | undefined {
+  if (!a) return b
+  if (!b) return a
+  if (isObject(a) && isObject(b)) {
+    return { ...a, ...b }
```

Comment:
The order here isn't important?

Reply:
B should override A, which is the behavior of object merging, but not array previously